tooling/bin/assess-minibroker-charts

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
+set -o errexit -o nounset
+
+# Location .............................................................
+
+: "${GIT_ROOT:=$(git rev-parse --show-toplevel)}"
+cd "${GIT_ROOT}"
+
+CAP=changeme
+CF_NAMESPACE=cf
+INCREMENTAL=0
+MODE="fresh, clearing previous data"
+
+function usage() {
+    cat <<EOF
+Usage: $(basename "${0}") [options]
+
+  -h:      Displays this help message
+  -p PASS: Specified cluster admin password
+  -n NAME: Specifies cluster namespace, default: cf
+  -i:      Activates incremental mode which skips processed charts
+EOF
+}
+
+while getopts "hp:n:i" opt; do
+    case $opt in
+	h)  usage
+	    exit
+	    ;;
+	p)  CAP=
+	    ;;
+	n)  CF_NAMESPACE=
+	    ;;
+	i)  INCREMENTAL=1
+	    MODE="incremental, keeping previous data"
+	    ;;
+    esac
+done
+
+TOP=$(pwd)
+source "${TOP}/make/include/colors"
+
+echo $(cyan Configuration):
+echo '  - Namespace:' $(blue "${CF_NAMESPACE}")
+echo '  - Password: ' $(blue "${CAP}")
+echo '  - Mode:     ' $(blue "${MODE}")
+
+# General operation
+# 1. Get the master index from the stable helm repository at google.
+# 2. Get the entries for our supported databases out of the master.
+#    a. For all engines, get a list of (engine-version, chart-version,
+#       location) tuples out of the files from the previous step.
+#    b. For all engines, and all their tuples, create a proper helm index
+#       referencing exactly that engine and chart.
+# 3. For all engines, and all their tuples, test.
+
+# Configuration ........................................................
+
+engines=""
+engines="${engines} mariadb"
+engines="${engines} mongodb"
+engines="${engines} postgresql"
+engines="${engines} redis"
+# Add new engines here, and at (x).
+
+bin="${TOP}/tooling/mb-chart-assessment"
+wrk="${TOP}/_work"
+stem=kube-chart-index
+rstem=dbe
+master="${wrk}/${rstem}-charts-master-index.yaml"
+
+stable=https://kubernetes-charts.storage.googleapis.com
+index="${stable}/index.yaml"
+
+nodeenv=src/scf-release/src/acceptance-tests-brain/test-resources/node-env
+
+# Setup ................................................................
+
+if [ $INCREMENTAL -eq 0 ] ; then
+    echo $(cyan Cleared) previous run, starting from empty
+    mkdir -p "${wrk}"
+    # Clear leftovers from previous runs
+    rm -rf ${wrk}/${rstem}-*
+else
+    echo $(cyan Incremental) run, keeping data, will skip processed charts
+fi
+
+# 1. Get master index .....................................................
+
+echo $(cyan Retrieving) master index ...
+curl --silent "${index}" > "${master}"
+
+# 2. Extract the full data for all the supported engines ..................
+
+for engine in $engines
+do
+    echo -n $(cyan Extracting) engine $(blue "${engine}") ...
+    # Originally attempted to use a pipeline using `y2j | jq ...`.
+    # y2j failed on the incoming yaml, claiming a number of things as
+    # not serializable to json.
+    ruby "${bin}/get-charts-for-engine.rb" "${engine}" "${master}" \
```

Comment:
I started with shell and then ran into issues with using `y2j` (failed to parse the index), so my yaml-specific helpers became ruby.

Can we move a rewrite to ruby into a new ticket ?

Reply:
I'd rather keep it in this one, because it's basically a fresh code review anyway. I understand that you want to get something merged, just to have some progress, though…

Reply:
I am also not as firm in my ruby as well. Ok. Will make the small changes first, and then see how well I can make this working as a ruby.

Reply:
Superceded by ruby conversion (commit 
The script `local-chart-repository` determines the patch to use and
applies it as part of setting up a helm repository for a chart.
 and 2 preceding)